src/sentry/api/endpoints/api_tokens.py

```diff
@@ -119,3 +123,30 @@ def delete(self, request: Request):
         analytics.record("api_token.deleted", user_id=request.user.id)
 
         return Response(status=204)
+
+    @method_decorator(never_cache)
+    def put(self, request: Request) -> Response:
+        serializer = ApiTokenNameSerializer(data=request.data)
+
+        if serializer.is_valid():
+            result = serializer.validated_data
+
+            user_id = self._get_appropriate_user_id(request=request)
+            token_id = request.data.get("tokenId", None)
+
+            if token_id is None:
+                return Response({"tokenId": token_id}, status=400)
+
+            with outbox_context(transaction.atomic(router.db_for_write(ApiToken)), flush=False):
+                token_to_rename: ApiToken | None = ApiToken.objects.filter(
+                    id=token_id, application__isnull=True, user_id=user_id
+                ).first()
+
+                if token_to_rename is None:
+                    return Response({"tokenId": token_id, "userId": user_id}, status=400)
+
+                token_to_rename.update(name=result.get("name", None))
+                token_to_rename.save()
```

Comment:
`update` does the job for you, you don't need to call `save` again unless you want to invoke `post_save` or `pre_save` Django signals (or have some special save logic for the model) https://docs.djangoproject.com/en/5.0/topics/db/queries/#updating-multiple-objects-at-once

although we have some special save logic here https://github.com/getsentry/sentry/blob/7d2cb635fe65068d0a2f1d1fbb0aec106d80f6d1/src/sentry/models/apitoken.py#L209-L225

so we should call `save`. so i'd change this to

```
token_to_rename.name = result.get("name", None)
token_to_rename.save()
```